### PR TITLE
Pass config env vars to mirrord verify-config.

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: #70 

Passes the `launch.json` `env` section to `mirrord verify-config`, and `mirrord`, resolving config options that were set as env vars.

- intellij PR: [#181](https://github.com/metalbear-co/mirrord-intellij/pull/181)

### For the reviewer:

Think we're not using any standard formatter for this, so huge part of the diff is just indentation changes.

The [src/tests/e2e.ts](https://github.com/metalbear-co/mirrord-vscode/pull/74/files#diff-b43d278da30fffad8eb324e1f1165f5e2ae4ecddd858801a51e0ce3e08c7f65c) file was only changed to add a `;` that was missing.

Most of the changes have to do with `configEnv` or `config.env`.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -144,18 +169,19 @@ export class MirrordAPI {
...
 
   // Return environment for the spawned mirrord cli processes.
-  private static getEnv(): NodeJS.ProcessEnv {
+  private static getEnv(configEnv: EnvVars): NodeJS.ProcessEnv {
     // clone env vars and add MIRRORD_PROGRESS_MODE
     return {
       // eslint-disable-next-line @typescript-eslint/naming-convention
       "MIRRORD_PROGRESS_MODE": "json",
+      ...configEnv,
       ...process.env,
```
Comment (Razz4780) [inline src/api.ts:178]:
```suggestion
      ...process.env,
      ...configEnv,
      // eslint-disable-next-line @typescript-eslint/naming-convention
      "MIRRORD_PROGRESS_MODE": "json",
```

`...` operator will overwrite previous values, I think we want this order

File: src/debugger.ts [hunk 2]
```diff
@@ -14,231 +14,231 @@ const DYLD_ENV_VAR_NAME = "DYLD_INSERT_LIBRARIES";
...
+  let mirrordApi = new MirrordAPI(cliPath);
+
+  config.env ||= {};
+  let target = null;
+
+  let configPath = await MirrordConfigManager.getInstance().resolveMirrordConfig(folder, config);
+  const verifiedConfig = await mirrordApi.verifyConfig(configPath, config.env);
+
+  // If target wasn't specified in the config file (or there's no config file), let user choose pod from dropdown
+  if (!configPath || (verifiedConfig && !isTargetSet(verifiedConfig))) {
```
Comment (Razz4780) [inline src/debugger.ts:141]: When I set `"target": "targetless"` in the mirrord config, the target quickpick still pops up :(

**Review by Razz4780 — CHANGES_REQUESTED** [review]

File: src/debugger.ts (hunk 2 above)
Comment (meowjesty) [reply src/debugger.ts:141]: Both `target: targetless` and not specifying a target return the same thing from config:

```json
{
  "type": "Success",
  "config": {
    "path": null,
    "namespace": null
  },
  "warnings": []
}
```

We need to change it to be explicit I guess, something like:

```json
{
  "type": "Success",
  "config": {
    "path": "targetless",
    "namespace": null
  },
  "warnings": []
}
```

Comment (meowjesty) [reply src/debugger.ts:141]: Created an issue for this in mirrord [#2010](https://github.com/metalbear-co/mirrord/issues/2010), as I'm not sure we have a good of solving this here?

Comment (Razz4780) [reply src/debugger.ts:141]: Yeah, that should probably be resolved in the CLI

**Review by Razz4780 — APPROVED** [review]
:cat2:

**Pull request merged**